### DISCUSSION TITLE
Strange DOM interaction with an <h3>

### DISCUSSION THREAD

**Discussion by pgeipi10** [KaTeX/KaTeX#3231] [Q&A] [answered]
Why is the KaTeX output included in `<h3>` (see border) when DOM-wise it is clearly outside of it?

<img width="1427" alt="Screen Shot 2021-09-04 at 1 54 05 PM" src="https://user-images.githubusercontent.com/7796257/132104037-1a6ccbb4-acf8-456b-b3b4-cc8cd54c1931.png">

**Comment by edemaine** [comment]
That definitely looks weird. But it's pretty hard to tell what's going on from the screenshot alone. Could you post a link to the website, or a minimum reproduction?

**Reply by pgeipi10** [reply]
See section 12.6.3 [here](https://grinfeld.org/books/IntroductionToTensorCalculus/Chapter12.html).

**Comment by edemaine** [comment, marked as answer]
Ah, this is because of the float hack from the other thread. You'll want to add a `clear` style to something afterward to end the floats. For example, you could add the following:
```css
h3 { clear: both }
```
Or more robustly, you could add a "[clearfix](https://css-tricks.com/snippets/css/clear-fix/)" to `.katex-display`:
```css
.katex-display:after {
  content: "";
  display: table;
  clear: both;
}
```
The latter seems to be failing in your context because you're wrapping every `.katex-display` with a `<div>`. It works if you can take them outside the `<div>` somehow.

**Reply by pgeipi10** [reply]
Actually, I was going to ask about that. If I don't wrap my .katex-displays in a div, they started appearing out of order with text. I can demonstrate that if necessary.  Is that also a side-effect of the fix from the other thread? Is there a different fix for that?

**Reply by edemaine** [reply]
The clearfix fix above should fix that as well (hopefully).
